File: fyn-monolithic/src/main/java/com/fyn_monolithic/dto/request/post/CreatePostRequest.java
package com.fyn_monolithic.dto.request.post;

import com.fyn_monolithic.model.post.PostVisibility;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.Set;

@Data
public class CreatePostRequest {

    @Size(max = 2048)
    private String content; // Optional - can be null or empty

    private Set<String> hashtags;

    private Set<String> mentionUsernames;

    private PostVisibility visibility = PostVisibility.PUBLIC;
}

File: fyn-flutter-app/lib/features/post/data/models/create_post_request.dart
import 'post_visibility.dart';

class CreatePostRequest {
  final String? content;  // Optional - can be null or empty
  final Set<String>? hashtags;
  final Set<String>? mentionUsernames;
  final PostVisibility visibility;
  final double? latitude;
  final double? longitude;
  final String? placeCode;

  CreatePostRequest({
    this.content,  // Now optional
    this.hashtags,
    this.mentionUsernames,
    this.visibility = PostVisibility.public,
    this.latitude,
    this.longitude,
    this.placeCode,
  });

  Map<String, dynamic> toJson() {
    return {
      'content': content ?? '',  // Send empty string if null
      if (hashtags != null) 'hashtags': hashtags!.toList(),
      if (mentionUsernames != null) 'mentionUsernames': mentionUsernames!.toList(),
      'visibility': visibility.serverValue,
      if (latitude != null) 'latitude': latitude,
      if (longitude != null) 'longitude': longitude,
      if (placeCode != null) 'placeCode': placeCode,
    };
  }
}

